Mathlib/RingTheory/IntegralClosure/IntegrallyClosed.lean

```diff
@@ -275,3 +275,53 @@ theorem isIntegrallyClosedOfFiniteExtension [IsDomain R] [FiniteDimensional K L]
   (integralClosure_eq_bot_iff L).mp integralClosure_idem
 
 end integralClosure
+
+/-- Any field is integral closed. -/
+instance Field.instIsIntegrallyClosed (K : Type*) [Field K] : IsIntegrallyClosed K :=
+  (isIntegrallyClosed_iff K).mpr fun {x} _ ↦ ⟨x ,rfl⟩
+
+open Localization in
+/-- An integral domain `R` is integral closed if `Rₘ` is integral closed
+for any maximal ideal `m` of `R`. -/
+theorem IsIntegrallyClosed.of_localization_maximal {R : Type*} [CommRing R] [IsDomain R]
+    (h : ∀ p : Ideal R, p ≠ ⊥ → [p.IsMaximal] → IsIntegrallyClosed (Localization.AtPrime p)) :
+    IsIntegrallyClosed R := by
+  by_cases hf : IsField R
+  · exact (IsField.toField hf).instIsIntegrallyClosed
+  apply (isIntegrallyClosed_iff (FractionRing R)).mpr
+  rintro ⟨x⟩ hx
+  let I : Ideal R := {
```

Comment:
Is it possible to get `I` with current technology, ie without defining it explicitly?

Reply:
I have changed the definition of `I` to `Ideal.span {x.2.1} / Ideal.span {x.1}`. This brought some technical difficulties regarding localization, so I have added several lemmas about Localization.